src/js/components/DataTable/Header.js

```diff
@@ -347,7 +373,11 @@ const Header = forwardRef(
               }
 
               // content should fill any available space in cell
-              content = <Box flex="grow">{content}</Box>;
+              content = (
+                <Box flex="grow" fill={onResize ? 'vertical' : false}>
```

Comment:
Can you add a comment of why we only do `fill` when it's onResize?

Reply:
I'm not sure I follow why

Reply:
I added a comment, I also realized this should be `fill={onResize || search ? 'vertical' : false}` so I update this in the latest commit. The reason `onResize` and `search` need the `fill` prop is later in the code if `onResize` or `search` is true content will get wrapped with another Box (see line 398 in Header.js), they need to be wrapped with another Box to have the correct behavior when resizing or typing in the search. If we don't set `fill` on the original Box it won't expand to fill the full space if wrapped with another Box.

Reply:
Your new comment looks great helps me understand